cpp/ql/src/Diagnostics/LinesOfCode.ql

```diff
@@ -0,0 +1,10 @@
+/**
+ * @name Total lines of code
+ * @description The total number of lines of code across all source files.
+ * @kind metric
+ * @id cpp/metrics/lines-of-code
+ */
+
+import cpp
+
+select sum(File f | f.fromSource() | f.getMetrics().getNumberOfLinesOfCode())
```

Comment:
For the benefit of anyone else who is wondering, `getNumberOfLinesOfCode()` doesn't count blank lines and lines that only contain comments.